scandium-core/src/main/java/org/eclipse/californium/scandium/config/DtlsConnectorConfig.java

```diff
@@ -2057,6 +2154,15 @@ public DtlsConnectorConfig build() {
 			if (config.useHandshakeStateValidation == null) {
 				config.useHandshakeStateValidation = Boolean.TRUE;
 			}
+			if (config.useTruncatedCertificatePathForClientsCertificateMessage == null) {
+				config.useTruncatedCertificatePathForClientsCertificateMessage = Boolean.TRUE;
+			}
+			if (config.useTruncatedCertificatePathForValidation == null) {
+				config.useTruncatedCertificatePathForValidation = Boolean.TRUE;
+			}
+			if (config.useKeyUsageVerification == null) {
+				config.useKeyUsageVerification = Boolean.TRUE;
+			}
```

Comment:
Half true. Though the tests are more complete now (not only client handshaker on own certificate), the true enables now more checking. For me this is intended, and if someone doesn't want the key-usage check, it may be disabled at all.

Reply:
Like said in https://github.com/eclipse/californium/issues/1166#issuecomment-569504207.
Behavior break is not desirable for a 2.1.
But as I said in my email privately, I just notice this kind of issue but not insist on it anymore.